util/stop_watch.h

```diff
@@ -101,36 +101,83 @@ class StopWatch {
   const uint64_t start_time_;
 };
 
+class SimpleStopWatch {
+ public:
+  bool IsStarted() { return start_ != 0; }
+  explicit SimpleStopWatch(SystemClock* clock) : clock_(clock), start_(0) {}
+  virtual void Start() = 0;
+  virtual uint64_t Elpased() = 0;
+  virtual ~SimpleStopWatch() = default;
+
+ protected:
+  SystemClock* clock_;
+  uint64_t start_;
+};
+
 // a nano second precision stopwatch
 class StopWatchNano {
  public:
-  explicit StopWatchNano(SystemClock* clock, bool auto_start = false)
-      : clock_(clock), start_(0) {
+  explicit StopWatchNano(SystemClock* clock, bool auto_start = false,
+                         bool cpu_time = false)
+      : clock_(clock), start_(0), is_time_type_cpu_(cpu_time) {
     if (auto_start) {
       Start();
     }
   }
-
-  void Start() { start_ = clock_->NowNanos(); }
-
+  void Start() {
+    if (is_time_type_cpu_) {
```

Comment:
This conditional branch can create potentially significant overhead at the nanosecond scale. Hopefully the compiler would optimize, but I wouldn't depend on it for good results when it's avoidable. (See next comment.)